src/main/java/seedu/address/storage/XmlAdaptedGameManager.java

```diff
@@ -0,0 +1,111 @@
+package seedu.address.storage;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Objects;
+import javax.xml.bind.annotation.XmlElement;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.game.DecreasingMode;
+import seedu.address.model.game.FlatMode;
+import seedu.address.model.game.GameManager;
+import seedu.address.model.game.GameMode;
+import seedu.address.model.game.IncreasingMode;
+import seedu.address.model.game.PriorityMode;
+
+/**
+ * JAXB-friendly version of the {@code GameManager}.
+ */
+public class XmlAdaptedGameManager {
+
+    @XmlElement(required = true)
+    private String gameMode;
+
+    @XmlElement
+    private String period;
+
+    @XmlElement
+    private String lowXp;
+
+    @XmlElement
+    private String highXp;
+
+    /**
+     * Constructs an {@code XmlAdaptedAchievementRecord}.
+     * This is the no-arg constructor that is required by JAXB.
+     */
+    public XmlAdaptedGameManager() {}
+
+    /**
+     * Converts a given AchievementRecord into this class for JAXB use.
+     *
+     * @param source future changes to this will not affect the created {@code XmlAdaptedAchievementRecord}.
+     */
+    public XmlAdaptedGameManager(GameManager source) {
+        requireNonNull(source);
+
+        GameMode mode = source.getGameMode();
+
+        gameMode = mode.getClass().getSimpleName();
```

Comment:
LoD violation?

Reply:
I don't think it's a violation / a violation in spirit -- getClass is a java thing which doesn't change and so it doesn't reflect internal structure. But anyway this line of code has been rewritten out now so it won't really matter.